docs/2.7.5/docs/query/pqs-user-guide.rst

```diff
@@ -175,52 +175,197 @@ and Operators <https://www.postgresql.org/docs/12/functions-json.html>`__ in
+
+Although the figure shows integers as an offset value, this may change in the future. The format of offsets should be treated as opaque to the client. No client-side transformation of an offset (such as subtracting or adding one offset to another and using that in a ledger read)  is guaranteed to return a meaningful offset. However, you can always expect that offsets are lexicographically comparable.
+
+A transaction ID compares to an offset in the following ways:
+ 
+ - Not every offset has a transaction ID. For example, the completion event of a rejected transaction won't have a transaction ID because the transaction did not successfully complete.
+ - There is, at most, one transaction ID at a single offset.
+ - Every accepted transaction is published at a single offset. 
+ - Offsets are local to a participant, whereas transaction IDs are virtual ledger-wide identifiers to correlate transactions across multiple participants.
+ - Offsets can establish a temporal ordering within the same sync domain ("happened-before" relationship) between transactions and/or events from different transactions.
+
+For analysis of a participant node's ledger, the offset is a better identifier than transaction ID because it can be used for ordering and an offset range can be specified to limit the analysis. However, if the analysis must coordinate with other participant nodes, then transaction IDs are better. 
+
+If a participant node's ledger is restored from a backup, it will review the data on the sequencer and process it to the latest information. However, in performing this rehydration, the resulting PN can have a different set of offset values than it had before the backup. For Daml 2.x,, the order of event processing and transaction IDs remain the same, but the resulting offsets likely change. If a client application caches offset data or stores it in a database, those datastores will also need to be replayed to have the proper offset values of the restored participant node.
+
+.. _pqs-time-model:
+
+Time Model within PQS
 =====================
 
+PQS builds on the offset concept of the participant node's ledger, and PQS is a valid representation of that ledger. PQS processes events asynchronously and concurrently, but the PQS programming model is intended to simplify development. The offset forms the basis for the PQS programming model. Helper functions make it easy to reason about offset values, pushing that complexity into the background. This section discusses how PQS models time.
+
+NOTE:  Use of the helper functions is required for your application to upgrade in future releases. Direct table access is unsupported because the schema may change in a future release. There should be no inherent overhead in using the API so performance should not suffer.
```

Comment:
@hrischuk-da Does this need further explanation? The way I read it, you need to use the helper functions now. If you directly access the table instead of using the helper functions and the schema changes in an upgrade, you might run into problems? I could be wrong.

Reply:
@loiskilberg-da your reading is correct.  Please feel free to embellish my chicken scrath.